packages/react-admin/src/mui/list/DatagridBody.js

```diff
@@ -27,7 +28,8 @@ const DatagridBody = ({
     >
         {ids.map((id, rowIndex) => (
             <TableRow
-                style={rowStyle ? rowStyle(data[id], rowIndex) : styles.tr}
+                className={classes.row}
```

Comment:
I think you should add a `rowEven` and `rowOdd` classes, that the user could override in `classes`